### ISSUE TITLE
NuGet cli install with -DependencyVersion Lowest parameter not working as expected

### ISSUE THREAD

**Issue by jc1001** [NuGet/Home#8651]
## Details about Problem

NuGet product used: NuGet.exe
NuGet version: 5.2.0
OS version: Windows 10, Version 1903
Worked before? Not as far as I can tell

## Detailed repro steps so we can see the same problem

I have a NuGet package with 2 dependencies:
```xml
<dependencies>
      <dependency id="YamlDotNet.Signed" version="5.1.0" />
      <dependency id="Newtonsoft.Json" version="8.0.1" />
</dependencies>
```
Both of these are valid dependencies that have higher versions available.

When I install my package (using the NuGet cli tool) , I use the following syntax:
```
nuget.exe install MyPackage -Source c:\Temp\Packages -Source https://api.nuget.org/v3/index.json -DependencyVersion Lowest -OutputDirectory c:\Temp\Output
```
In this case, I would expect the output directory (c:\Temp\Output) to contain the following packages:
```
MyPackage.1.0.0
Newtonsoft.Json.8.0.1
YamlDotNet.Signed.5.1.0
```

It however contains the following:
```
MyPackage.1.0.0
Newtonsoft.Json.8.0.1
YamlDotNet.6.0.0
YamlDotNet.Signed.6.0.0
```
### Verbose Logs
```
NuGet Version: 5.2.0.6090
Feeds used:
  c:\Temp\Packages
  https://api.nuget.org/v3/index.json

Installing package 'MyPackage' to 'c:\Temp\Output'.
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/mypackage/index.json


Attempting to gather dependency information for package 'MyPackage.1.0.0' with respect to project 'c:\Temp\Output', targeting 'Any,Version=v0.0'
Package MyPackage from source c:\Temp\Packages gathered from cache.
Package MyPackage from source nuget.org gathered from cache.
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet.signed/index.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet.signed/page/3.3.1/3.9.0-pre262.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet.signed/page/3.9.0-pre263/4.2.2-net200423.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/newtonsoft.json/index.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet.signed/page/4.2.2-pre0424/6.0.0.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/system.collections/index.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet/index.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet/page/2.3.0-rc/3.9.0-pre239.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet/page/3.9.0-pre240/4.1.1-pre0355.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/yamldotnet/page/4.1.1-pre0356/8.0.0-emit-default-values0565.json
  CACHE https://api.nuget.org/v3/registration3-gz-semver2/system.runtime/index.json
Total number of results gathered : 17
Gathering dependency information took 201.43 ms
Summary of time taken to gather dependencies per source :
c:\Temp\Output	-	11.53 ms
https://api.nuget.org/v3/index.json	-	289.18 ms
c:\Temp\Packages	-	11.41 ms
Attempting to resolve dependencies for package 'MyPackage.1.0.0' with DependencyBehavior 'Lowest'
Resolving dependency information took 0 ms
Resolving actions to install package 'MyPackage.1.0.0'
Resolved actions to install package 'MyPackage.1.0.0'
Retrieving package 'MyPackage 1.0.0' from 'c:\Temp\Packages'.
Retrieving package 'Newtonsoft.Json 8.0.1' from 'nuget.org'.
Retrieving package 'YamlDotNet 6.0.0' from 'nuget.org'.
Retrieving package 'YamlDotNet.Signed 6.0.0' from 'nuget.org'.
Adding package 'Newtonsoft.Json.8.0.1' to folder 'c:\Temp\Output'
Added package 'Newtonsoft.Json.8.0.1' to folder 'c:\Temp\Output'
Added package 'Newtonsoft.Json.8.0.1' to folder 'c:\Temp\Output' from source 'https://api.nuget.org/v3/index.json'
Successfully installed 'Newtonsoft.Json 8.0.1' to c:\Temp\Output
Adding package 'YamlDotNet.6.0.0' to folder 'c:\Temp\Output'
Added package 'YamlDotNet.6.0.0' to folder 'c:\Temp\Output'
Added package 'YamlDotNet.6.0.0' to folder 'c:\Temp\Output' from source 'https://api.nuget.org/v3/index.json'
Successfully installed 'YamlDotNet 6.0.0' to c:\Temp\Output
Adding package 'YamlDotNet.Signed.6.0.0' to folder 'c:\Temp\Output'
Added package 'YamlDotNet.Signed.6.0.0' to folder 'c:\Temp\Output'
Added package 'YamlDotNet.Signed.6.0.0' to folder 'c:\Temp\Output' from source 'https://api.nuget.org/v3/index.json'
Successfully installed 'YamlDotNet.Signed 6.0.0' to c:\Temp\Output
Adding package 'MyPackage.1.0.0' to folder 'c:\Temp\Output'
Added package 'MyPackage.1.0.0' to folder 'c:\Temp\Output'
Added package 'MyPackage.1.0.0' to folder 'c:\Temp\Output' from source 'c:\Temp\Packages'
Successfully installed 'MyPackage 1.0.0' to c:\Temp\Output
Executing nuget actions took 533.56 ms
```
### Sample Project

I've attached a zip file that contains the directory structure and necessary files that I've used in the example above.
[Temp.zip](https://github.com/NuGet/Home/files/3676995/Temp.zip)

Thank you and please let me know if you require anything else.

**Comment by rrelyea** [comment]
The .nuspec file inside of YamlDotNet.Signed lists:
```
    <dependencies>
      <dependency id="YamlDotNet" version="6.0.0" />
    </dependencies>
```
NuGet Install tries to download all dependencies. Therefore YamlDotNet is one of the packages downloaded.

**Comment by jc1001** [comment]
Thank you for the reply.  You are correct in that the latest version of YamlDotNet.Signed (v6.0.0) does have a dependency on YamlDotNet 6.0.0.  

In my test package above, I am however installing YamlDotNet.Signed 5.1.0 which does not have any dependencies.  

```xml
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
  <metadata minClientVersion="2.8">
    <id>YamlDotNet.Signed</id>
    <version>5.1.0</version>
    <authors>Antoine Aubry</authors>
    <owners>Antoine Aubry</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <licenseUrl>https://github.com/aaubry/YamlDotNet/blob/master/LICENSE</licenseUrl>
    <projectUrl>https://github.com/aaubry/YamlDotNet/wiki</projectUrl>
    <iconUrl>http://aaubry.net/images/yamldotnet.png</iconUrl>
    <description>A .NET library for YAML. YamlDotNet provides low level parsing and emitting of YAML as well as a high level object model similar to XmlDocument.</description>
    <summary>This package contains the YAML parser and serializer. This is the signed version of the package.</summary>
    <language>en-US</language>
    <tags>yaml parser development library serialization signed strongname</tags>
    <repository type="git" url="https://github.com/aaubry/YamlDotNet.git" />
  </metadata>
</package>
```
Since I am using the NuGet cli with the -DependencyVersion Lowest parameter (see original issue above), I would expect:
a) that the output directory contains YamlDotNet.Signed 5.1.0, not 6.0.0 and
b) that YamlDotNet 6.0.0 is not present (since it is not a dependency of 5.1.0)

I've attached the 2 versions of YamlDotNet.Signed that are applicable to this issue:
[Packages.zip](https://github.com/NuGet/Home/files/3701345/Packages.zip)

**Comment by jc1001** [comment]
Hi @rrelyea, 
Can you please look at this again if possible?  The answer that you gave me, although correct, was for a different version of the NuGet package than the one that I was trying to install.  I did confirm that the original problem (as desribed above) is still present in the latest version of NuGet.exe (version 5.3).

**Comment by nkolev92** [comment]
@v-luzh can you please try to repro this? 
Thank you.

**Comment by v-luzh** [comment]
Hi @nkolev92, Sorry, I missed the comment. 
It is fixed on VS Main\31820.317 + NuGet Client Dev\6.1.0.30 as below screenshot. I think it is OK to close this bug now.
![image](https://user-images.githubusercontent.com/20632625/138686330-93a21ebb-6a6f-45f8-8c25-c09edfdb238d.png)

**Comment by nkolev92** [comment]
Thanks @v-luzh and no worries :)

Closing per the above comment.
